src/storage/btree.c

```diff
@@ -5618,11 +5618,17 @@ xbtree_add_index (THREAD_ENTRY * thread_p, BTID * btid, TP_DOMAIN * key_type, OI
 
   if (unique_pk)
     {
-      root_header->num_oids = num_oids;
-      root_header->num_nulls = num_nulls;
-      root_header->num_keys = num_keys;
+      root_header->num_oids = num_oids & 0xffffffff;
+      root_header->num_nulls = num_nulls & 0xffffffff;
+      root_header->num_keys = num_keys & 0xffffffff;
       root_header->unique_pk = unique_pk;
 
+      over += root_header->_64.num_oids = num_oids >> 32;
+      over += root_header->_64.num_nulls = num_nulls >> 32;
+      over += root_header->_64.num_keys = num_keys >> 32;
```

Comment:
Can't 'over' overflow and be negative?

Reply:
num_x's uses 42 bits of 64 bits, so the overflown is not possible.

Reply:
Where is the guarantee that the value of the num_xxxs variables passed as parameters is within the 42-bit size?

Reply:
it's a 64 bit number uses 42-bit only.